department-of-veterans-affairs/caseflow
# removed judge check from affinity filter

Resolves [APPEALS-58200](https://jira.devops.va.gov/browse/APPEALS-58200)

# Description
Removed judge check from affinity filter in docket.rb
Updated affinity filter to allow for nil judge queries
Added not_affinity param to some methods that should not filter by affinity

Updated case_docket.rb to add the lever affinity filters to genpop_priority_count
Created custom method to filter genpop appeals without needing a judge
Updated case_docket_spec.rb to test for new genpop_priority_count changes

## Acceptance Criteria

- [ ] The job should not create distributions with zero cases. Judges with a target number of cases less than or equal to zero should be filtered out.

## Testing Plan

1. On local or demo from a reset state or fresh deploy:
2. Clear all ready appeals by running these commands in console:
• DistributionTask.where(status: 'assigned')&.map(&:on_hold!)
• VACOLS::Case.where(bfcurloc: ['81', '83'])&.map { |c| c.update!(bfcurloc: 'testing') }
3. Run the new seed files in the rails console or via the below rake command (NOT in a rails console):
• bundle exec rake db:seed:ama_affinity_cases
and/or
• bundle exec rake db:seed:cavc_affinity_levers_test_data

4. Download the ready_to_distribute CSV and verify that there are appeals within their affinity that should not be distributed when running the Push Priority Job.
5. Run the push priority job from the scheduled jobs tab in caseflow, and continue running until their are no more appeals.
6. Ensure the distribution_stats and distributions do not contain a batch size of 0.
7. Download the ready_to_distribute CSV to verify none of the appeals within their affinity were distributed.
